packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -187,7 +181,7 @@ function warnNoop(
   if (__DEV__) {
     const constructor = publicInstance.constructor;
     const componentName =
-      (constructor && getComponentName(constructor)) || 'ReactClass';
+      (constructor && getComponentName({type: constructor})) || 'ReactClass';
```

Comment:
This part is kinda hacky and is the reason they weren’t shared. It’s just a different data structure. Maybe it’s reasonable to change the function itself to accept the type directly, but I would rather have duplication than “lie” about the type like here.

Reply:
Yes I had the hacky reservation while making the change. I’m open to changing the shared function to just accept the type or I can close this PR. Doesn’t bother me either way, just let me know how you want to proceed.

Reply:
Let’s just keep it how it is? I don’t see us changing this function anyway so it’s not worth unifying.